### ISSUE TITLE
M04-LAB10: Instructions tell to use a "previously created Service Principal" never created in the lab

### ISSUE THREAD

**Issue by massimobonanni** [MicrosoftLearning/AZ400-DesigningandImplementingMicrosoftDevOpsSolutions#615]
### Contact Details

massimo.bonanni@tiscali.it

### What should we change?

I found a reference to a Service Principal creation in the lab M03-L05 ("Implement GitHub Actions for CI/CD"). I suggest to add same instructions to create the principal also in this lab. If the student make this lab without making the previous one, he/she cannot find the service principal and he/she has no idea how to create

### Lab

Lab 10 Integrate Azure Key Vault with Azure DevOps

### Relevant screenshots

paste here 😉

**Comment by LuizMacedo** [comment]
Thanks @massimobonanni. I'll review it since we are no longer using Service Principals. I may have missed something.

**Comment by LewisGoDeploy** [comment]
I have experienced the same issue in "Task 1: Setup and Run CI pipeline" 

The error was:
There was a resource authorization issue: "The pipeline is not valid. Job Build: Step AzureResourceManagerTemplateDeployment input ConnectedServiceName references service connection azure subs which could not be found. The service connection does not exist, has been disabled or has not been authorized for use. For authorization details, refer to https://aka.ms/yamlauthz. Job Build: Step DockerCompose1 input azureSubscriptionEndpoint references service connection azure subs which could not be found. The service connection does not exist, has been disabled or has not been authorized for use. For authorization details, refer to https://aka.ms/yamlauthz."

... indicating that the pipeline is trying to use a service connection named "azure subs", but that connection doesn't exist, maybe steps will be added to the guide to create this in future?

**Comment by LuizMacedo** [comment]
@massimobonanni @LewisGoDeploy that instruction will be updated to use the user email/name instead of Service Principal. I replaced Service Principals by Workload Identity federation in the [Validate lab environment](https://microsoftlearning.github.io/AZ400-DesigningandImplementingMicrosoftDevOpsSolutions/Instructions/Labs/AZ400_M00_Validate_lab_environment.html), "Create a service connection to access Azure resources".

**Comment by LewisGoDeploy** [comment]
@LuizMacedo Will this instruction for "Create a service connection to access Azure resources" need to be added to all individual modules? When this lab moves away from Azure pass?

**Comment by LuizMacedo** [comment]
@LewisGoDeploy no, the Validate Lab Environment page should be the very first one executed as a prerequisite. There’s no need to duplicate it in every lab. Make sure it is placed before Lab 1.

**Comment by massimobonanni** [comment]
I agree, but in every lab there is an Exercise 0 that, actually, is a repetition of the Validate Lab. Or we remove the exercise 0 from every labs or we add the service connection on every Exercise 0 on every lab

**Comment by LuizMacedo** [comment]
@massimobonanni Yes, all lab hosters should remove Exercise 0 and the steps for removing resources at the end (usually Exercise 02 or 03, depending on the lab). This has already been removed in Skillable, and others should do the same. The full steps in GitHub are intended for self-paced learners, and lab hosters should update their steps accordingly. Now, part of the environment should be auto provisioned, such as the Azure DevOps organization, team project, and import of the eShopOnWeb repository, etc.

**Comment by massimobonanni** [comment]
I understand, but if I am a self-paced student and I start the Lab10 as the first lab, there is no "service principal" previously created because there isn't any instruction that tells me to make the Lab 00 before the lab 10. At the beginning, there is "Exercise 0: (skip if done) Configure the lab prerequisites" and it doesn't mean yo need to go to lab00 and make the lab validation. A student can follow the instruction in the lab 10 without making the lab00. Probably, I miss something.

**Comment by LuizMacedo** [comment]
@massimobonanni Yes, I agree with you, and I’ll probably add the **Validate Lab Environment** step in the **Lab Requirements** section to make it clear that it cannot be skipped.

As I mentioned above, **Lab 10** will be updated. We no longer use any service principal in this new update for the service connections.

**Comment by sbonds** [comment]
A similar issue exists in https://github.com/sbonds/AZ400-DesigningandImplementingMicrosoftDevOpsSolutions/blob/6e122d10966ab2b68cca511d1ef642933ad91c98/Instructions/Labs/AZ400_M08_L14_Monitor_Application_Performance_with_Azure_Load_Testing.md with references to a previous Service Connection which is not part of the lab Exercise 0.

I don't envy you all the task of making this content useful under so many different circumstances. :-)
